sympy/physics/mechanics/tests/test_actuator.py

```diff
@@ -849,3 +853,157 @@ def test_to_loads(self):
                 diff = (calculated_component - expected_component).subs(substitutions).evalf()
                 # Check if the absolute value of the difference is below a threshold
                 assert Abs(diff) < 1e-9, f"The forces do not match. Difference: {diff}"
+
+class TestCoulombKineticFriction:
+    @pytest.fixture(autouse=True)
+    def _block_on_surface(self):
+        r"""A block sliding on a surface.
+
+        Notes
+        =====
+        This test validates the correctness of the CoulombKineticFriction by simulating
+        a block sliding on a surface with the Coulomb kinetic friction force.
+        The test covers scenarios with both positive and negative velocities.
+
+        Equations of Motion
+        ===================
+        The equations of motion for the block are derived considering
+        the applied force F, the mass m, the gravitational constant g, and the friction forces
+        as described above.
+
+        The general form of the equation of motion is:
+
+            F - F_friction = m * xdd
+
+        where F_friction is given by the friction force equation f_f(v).
+
+        """
+        # Define Dynamics symbols and parameters
+        self.q1, self.u1 = dynamicsymbols('q1 u1')
+
+        # Mass, gravity constant, friction coefficient, coefficient of sliding friction, viscous_coefficient
+        self.m, self.g, self.mu_k, self.mu_s, self.v_s, self.sigma, self.F = symbols('m g mu_k mu_s v_s sigma F', real=True)
+
+        # Define the reference frame
+        self.N = ReferenceFrame('N')
+        self.O = Point('O')
+        self.P = self.O.locatenew('P', self.q1 * self.N.x)
+        self.O.set_vel(self.N, 0)
+        self.P.set_vel(self.N, self.u1 * self.N.x)
+        self.pathway = LinearPathway(self.O, self.P)
+
+        self.particle = Particle('particle', self.P, self.m) # Block
+
+    def test_block_on_surface_default(self):
+        friction = CoulombKineticFriction(
+            self.mu_k,
+            self.m * self.g,
+            self.pathway,
+        )
+        system = System(self.N, self.O)
+        system.q_ind = [self.q1]
+        system.u_ind = [self.u1]
+        system.kdes = [self.q1.diff() - self.u1]
+        system.add_bodies(self.particle)
+        system.add_actuators(friction)
+        system.apply_uniform_gravity(-self.g * self.N.y)
+        system.add_loads(Force(self.particle, self.F * self.N.x))
+
+        eoms = system.form_eoms()
+
+        # Positive velocity case
+        positive_velocity = {self.u1: 3.141, self.q1: Abs(self.q1)}
```

Comment:
Can't you test it without depending on `q1` to be either positive or negative?

Reply:
Since 'simplify(eoms[0].xreplace({self.u1: 3.141}))' results in the piecewise expression 'Piecewise((F, Eq(sqrt(q1(t)**2)/q1(t), 0)), (F + g*m*mu_k/Abs(sqrt(q1(t)**2)/q1(t)), True))', I tried to specify `q1` so that we can use `sign` as expected.

Reply:
Can you make `q1, u1` real?